molvis/molvisbase/include/inviwo/molvisbase/datastructures/molecularstructure.h

```diff
@@ -151,6 +173,29 @@ namespace molvis {
  */
 class IVW_MODULE_MOLVISBASE_API MolecularStructure {
 public:
+    /**
+     * \brief data structure holding the atom indices of a single backbone segment including
+     * dihedral angles
+     */
+    struct BackboneSegment {
+        bool valid() const {
+            return ca.has_value() && n.has_value() && c.has_value() && o.has_value();
+        }
+
+        size_t resId;
+        size_t chainId;
+
+        std::optional<size_t> ca;
+        std::optional<size_t> n;
+        std::optional<size_t> c;
+        std::optional<size_t> o;
+
+        // dihedral angles
+        double phi = 0.0;
```

Comment:
Should these not be calculated then? What invariants you you have

Reply:
maybe angles optional